test/yagna/module/ya-provider/test_provider_multi_activity.py

```diff
@@ -0,0 +1,124 @@
+from goth.runner.provider import ProviderProbeWithLogSteps
+from goth.runner.requestor import RequestorProbeWithApiSteps
+
+from test.yagna.helpers.negotiation import negotiate_agreements, DemandBuilder
+from test.yagna.helpers.payment import pay_all
+
+logger = logging.getLogger(__name__)
+
+
+def _topology(
+    assets_path: Path, payment_id_pool: PaymentIdPool
+) -> List[YagnaContainerConfig]:
+    """Define the topology of the test network."""
+
+    # Nodes are configured to communicate via proxy
+    provider_env = node_environment(
+        rest_api_url_base=YAGNA_REST_URL.substitute(host=PROXY_HOST),
+    )
+    requestor_env = node_environment(
+        rest_api_url_base=YAGNA_REST_URL.substitute(host=PROXY_HOST),
+    )
+
+    provider_volumes = {
+        assets_path
+        / "provider"
+        / "presets.json": "/root/.local/share/ya-provider/presets.json"
+    }
+
+    return [
+        YagnaContainerConfig(
+            name="requestor",
+            probe_type=RequestorProbeWithApiSteps,
+            volumes={assets_path / "requestor": "/asset"},
+            environment=requestor_env,
+            payment_id=payment_id_pool.get_id(),
+        ),
+        YagnaContainerConfig(
+            name="provider_1",
+            probe_type=ProviderProbeWithLogSteps,
+            environment=provider_env,
+            payment_id=payment_id_pool.get_id(),
+            volumes=provider_volumes,
+        ),
+    ]
+
+
+# Tests running multiple activities on single Provider.
+# In this case Requestor is responsible for terminating Agreement.
+# Provider should listen
+@pytest.mark.asyncio
+async def test_provider_multi_activity(
+    assets_path: Path,
+    demand_constraints: str,
+    exe_script: dict,
+    payment_id_pool: PaymentIdPool,
+    runner: Runner,
+    task_package_template: str,
+):
+    """Test provider handling multiple activities in single Agreement."""
+
+    async with runner(_topology(assets_path, payment_id_pool)):
+        requestor = runner.get_probes(probe_type=RequestorProbeWithApiSteps)[0]
+        providers = runner.get_probes(probe_type=ProviderProbeWithLogSteps)
+
+        # Market
+        task_package = task_package_template.format(
+            web_server_addr=runner.host_address, web_server_port=runner.web_server_port
+        )
+
+        demand = (
+            DemandBuilder(requestor)
+            .props_from_template(task_package)
+            .property("golem.srv.caps.multi-activity", True)
+            .constraints(
+                "(&(golem.com.pricing.model=linear)\
+                (golem.srv.caps.multi-activity=true)\
+                (golem.runtime.name=wasmtime))"
+            )
+            .build()
+        )
+
+        agreement_providers = await negotiate_agreements(
+            requestor,
+            demand,
+            providers,
+        )
+
+        #  Activity
+        for agreement_id, provider in agreement_providers:
+            for i in range(0, 3):
```

Comment:
in another test we should check situtation when some1 tries to create more that one activity at a time

Reply:
Yes it's good idea. But I don't know if we know, what we expect from Provider in this case ;)